packages/core/core/src/worker.js

```diff
@@ -48,10 +48,13 @@ function loadOptions(ref, workerApi) {
 }
 
 async function loadConfig(cachePath, options) {
-  let processedConfig =
-    parcelConfigCache.get(cachePath) ??
-    nullthrows(await options.cache.get(cachePath));
-  let config = new ParcelConfig(
+  let config = parcelConfigCache.get(cachePath);
```

Comment:
Caching the actual ParcelConfig object instead of just its contents is important because it has a couple caches: one for loaded plugins, and another for compiled glob regexes. If we construct a new ParcelConfig each time, these are thrown away.